src/lava/magma/core/run_configs.py

```diff
@@ -67,42 +67,60 @@ def select(self,
         pass
 
 
-# ToDo: This is only a minimal RunConfig that will select SubProcModel if there
-#       is one and if not selects a PyProcessModel with the correct tag.
-#       Needs to be modified in future releases to support more complicated
-#       sync domains, @requires (example: GPU support), and to select
-#       LeafProcModels of type other than PyProcessModel.
 class Loihi1SimCfg(RunConfig):
     """Run configuration selects appropriate ProcessModel -- either
-    SubProcessModel for a Hierarchical Process or else a PyProcessModel for a
-    standard Process. The appropriate PyProcessModel is selected based @tag(
-    'floating_pt') or @tag('fixed_pt'), for floating point precision or Loihi
-    bit-accurate fixed point precision respectively"""
+    `SubProcessModel` for a hierarchical Process or else a `PyProcessModel`
+    for a standard Process.
 
-    def __init__(self, custom_sync_domains=None, select_tag='floating_pt',
-                 select_sub_proc_model=False):
+    If any `SubProcessModel`s are available, then they will be returned first.
+
+    In case of `PyProcessModel`s, an appropriate `PyProcessModel` is selected
+    based on the tags associated with it. Tags are set using the `@tag`
+    decorator. If no tags are set for `ProcessModel`s and no tag is selected,
+    then the first matching `ProcessModel` will be returned.
+
+    If any exceptions are needed, they can be specified in
+    `exception_proc_model_map` dictionary as `{Process: ProcessModel}`
+    key-value pairs. These explicit specifications are given
+    precedence over the subsequent logic for `ProcessModel` selection"""
+
+    def __init__(self,
+                 custom_sync_domains=None,
+                 select_tag=None,
+                 select_sub_proc_model=False,
+                 exception_proc_model_map=None):
         super().__init__(custom_sync_domains=custom_sync_domains)
         self.select_tag = select_tag
         self.select_sub_proc_model = select_sub_proc_model
+        self.exception_proc_model_map = exception_proc_model_map
+        if not exception_proc_model_map:
+            self.exception_proc_model_map = {}
 
     def select(self, proc, proc_models):
         from lava.magma.core.model.sub.model import AbstractSubProcessModel
         from lava.magma.core.model.py.model import AbstractPyProcessModel
-        py_proc_model = None
-        sub_proc_model = None
+        # First priority to the exceptions. We will simply return the
+        # ProcessModel class associated with a Process class in the
+        # exceptions dictionary
+        if proc.__class__ in self.exception_proc_model_map:
+            return self.exception_proc_model_map[proc.__class__]
+        # Now, we loop over all ProcessModels proc_models of a Process proc
         for pm in proc_models:
```

Comment:
Actually we made a mistake here:
- What we want is that if there's a SubProcModel, then we want to return it no matter what!
- But currently we are dependent on the order of the ProcModels provided!
- Even if we want the SubProcModel but the PyProcModel comes first, then we will return that!

Instead, we must not return immediately when we iterate through all the proc_models. We must first just store 'pm' as 'selected_pm' during the loop which starts out as None. If we first find a PyProcModel and later encounter a SubProcModel we may still overwrite 'selected_pm'.
Only when 'selected_pm' is still None in the end, do we raise the current AssertionError!

Reply:
This is done.